src/components/quiz.tsx

```diff
@@ -1,102 +1,139 @@
+import React, {FormEvent, MouseEventHandler, SyntheticEvent} from 'react'
 import {FunctionComponent} from 'react'
 import {Formik, Field, Form} from 'formik'
+import useShouldITrainMachine from 'hooks/use-should-i-train-machine'
+import {find} from 'lodash'
+import {Question as QuestionType} from 'utils/generate-question-states'
 
-let quiz = {
-  question: 'How many days in a row have you trained?',
-  answers: [
-    {
-      option: '4+ days',
-      value: '1',
-    },
-    {
-      option: '3 days',
-      value: '2',
-    },
-    {
-      option: '2 days',
-      value: '3',
-    },
-    {
-      option: '1 day',
-      value: '4',
-    },
-    {
-      option: 'Coming off a rest day',
-      value: '5',
-    },
-  ],
+function Quiz() {
+  const [current, send] = useShouldITrainMachine()
+
+  if (current.matches('success')) {
+    const responses = current.context.responses
+    const totalScore = responses.reduce((total, currentResponse) => {
+      return (total += currentResponse.score)
+    }, 0)
+    return (
+      <div className="text-center">
+        You are {totalScore * 4}% ready to train
+      </div>
+    )
+  }
+
+  const currentQuestion = current.context.currentQuestion
+
+  return <Question currentQuestion={currentQuestion} send={send} />
 }
 
-function Quiz({name = 'Quiz'}) {
+const defaultSelectedResponse = -1
+
+function Question({
+  currentQuestion,
+  send,
+}: {
+  currentQuestion: QuestionType
+  send: any
+}) {
+  const [selectedResponse, setSelectedResponse] = React.useState(
+    defaultSelectedResponse,
+  )
   return (
     <div className="grid grid-cols-6 gap-4 mt-8">
-      <h1 className="text-4xl leading-10 font-extrabold text-center col-start-2 col-span-4 sm:col-start-3 sm:col-span-2 mb-4">
-        {quiz.question}
+      <h1 className="text-4xl leading-10 font-extrabold text-center col-start-2 col-span-4 sm:col-start-3 sm:col-span-2">
+        {currentQuestion.value}
       </h1>
-      <Formik
-        initialValues={{[name]: ''}}
-        onSubmit={async (values) => {
-          await new Promise((r) => setTimeout(r, 500))
-          alert(JSON.stringify(values, null, 2))
+      {currentQuestion.byline && (
+        <h2 className="text-2xl text-gray-600 col-start-2 col-span-4 sm:col-start-3 sm:col-span-2 text-center">
+          ({currentQuestion.byline})
+        </h2>
+      )}
+      <form
+        className="col-start-2 col-span-4 sm:col-start-3 sm:col-span-2 mt-4"
+        onSubmit={(e) => {
+          e.preventDefault()
+          const currentResponse = find(currentQuestion.options, {
+            score: selectedResponse,
+          })
+          const eventData = {
+            selectedResponse: currentResponse?.value,
+            score: currentResponse?.score,
+            question: currentQuestion.value,
+            questionId: currentQuestion.id,
+          }
+          setSelectedResponse(defaultSelectedResponse)
+          send('CONFIRM', {data: eventData})
```

Comment:
when the user hits the button, we find the current response they selected to record to the machine context and then all of the data in `eventData`.